File: python/time_conversion.py
#!/bin/python3

import math
import os
import random
import re
import sys
from datetime import datetime

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#
"""
def timeConversion(s):
    if s[-2:] == "AM" and s[0:2] == "12":
        return "00" + s[2:-2]
    if s[-2:] == "PM" and s[0:2] != "12":
        return str(int(s[0:2]) + 12) + s[2:-2]
    
    return s[:-2]
"""

def timeConversion(s):
    return datetime.strptime(s, "%I:%M:%S%p").strftime("%H:%M:%S")

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    fptr.write(result + '\n')

    print(result)

    fptr.close()

File: cpp/time_conversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'timeConversion' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string timeConversion(string s) {
    if(
        s.compare(0, 2, "12") == 0 &&
        s.compare(8, 2, "AM") == 0
    ) {
        return "00" + s.substr(2, 6);
    }

    if(s.compare(8, 2, "PM") == 0 ) {
        if(s.compare(0, 2, "12") != 0 ) {
            return to_string(stoi(s.substr(0, 2)) + 12) + s.substr(2, 6);
        }
        else{
            return s.substr(0, 8);
        }
    }

    return s.substr(0, 8);
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
